utils/addressutils/utils.go

```diff
@@ -27,3 +30,12 @@ func DeriveAddress(addressType AddressType, moduleName, name string) sdk.AccAddr
 		return sdk.AccAddress{}
 	}
 }
+
+// Bech32ToAddr casts an arbitrary-prefixed bech32 string to either sdk.AccAddress or sdk.ValAddress.
+func Bech32ToAddr[T sdk.AccAddress | sdk.ValAddress](addr string) (T, error) {
```

Comment:
this could maybe be in the `sdk-utils` repo

Reply:
just saw your comment and copied code below. I agree, we should first migrate to v0.47